reviews/views.py

```diff
@@ -0,0 +1,66 @@
+import json, re, bcrypt, jwt
+
+from django.views           import View
+from django.http            import JsonResponse
+from django.db.models       import Avg, Q
+
+from reviews.models         import Review, ReviewImage
+from users.models           import User, WishList
+from users.utils            import login_decorator
+from restaurants.models     import Restaurant
+
+class ReviewView(View):
+    @login_decorator
+    def post(self, request, restaurant_id):
+        data    = json.loads(request.body)
+        user    = request.user
+
+        if Review.objects.filter(user_id = user.id, restaurant_id=restaurant_id).exists():
+            return JsonResponse({'MESSAGE':'REVIEW_EXIST'}, status=400)
+
+        review = Review.objects.create(
+            restaurant_id = restaurant_id,
+            user_id       = user.id,
+            description   = data['description'],
+            rating        = data['rating'],
+        )
+
+        ReviewImage.objects.create(
+            review_id = review.id,
+            image     = data['image'],
+        )
+
+        return JsonResponse({'MESSAGE':'SUCCESS'},status = 201)
+
+    @login_decorator
+    def patch(self, request, restaurant_id, review_id):
+        data = json.loads(request.body)
+        user = request.user
+
+        if not Review.objects.filter(id=review_id).exists():
+            return JsonResponse({'MESSAGE':'NOT_EXISTS'}, status=400)
+
+        Review.objects.filter(id=review_id).update(
+                restaurant_id = restaurant_id,
+                user_id       = user.id,
+                description   = data['description'],
+                rating        = data['rating'],
+                )
+
+        ReviewImage.objects.filter(review_id = review_id).update(
+            review_id = review_id,
+            image     = data['image'],
+            )
```

Comment:
잘 합쳐주셨네요!! 👍 

여기서 고려해야되는 두 가지만 더 고려하면 더 좋은 코드가 될 것 같습니다.
1. db에서 review를 조회할 때, user_id도 함께 조회한다. --> review_id로만 조회하면 누구든지 인증만 되면 어떤 리뷰도 업데이트할 수 있는 디버깅하기 힘든 오류가 발생할 수 있다.
2. line 50~ 53 로직은 예를들어 `review_id=1`에 해당하는 모든 image를 업데이트해버리는 오류가 있습니다.

두가지를 고려해서 코드를 수정해주세요!

```suggestion
        review = Review.objects.get(id=review_id, user_id=user.id)
        review.description = data['description']
        review.rating = data['rating']

        Review.objects.filter(id=review_id).update(
                restaurant_id = restaurant_id,
                user_id       = user.id,
                description   = data['description'],
                rating        = data['rating'],
                )

        ReviewImage.objects.filter(review_id = review_id).update(
            review_id = review_id,
            image     = data['image'],
            )
```

Reply:
아하! 그렇군요 ㅎㅎ 
그렇다면 filter(id=review_id, user_id=user.id)로 user_id 값도 함께 조회될 수 있도록 변경하겠습니다!